lib/Secrets/healthcheckrequest.cpp

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (C) 2018 Jolla Ltd.
+ * Contact: Timur Kristóf <timur.kristof@jollamobile.com>
+ * All rights reserved.
+ * BSD 3-Clause License, see LICENSE.
+ */
+
+#include "Secrets/healthcheckrequest.h"
+#include "Secrets/healthcheckrequest_p.h"
+
+#include "Secrets/secretmanager.h"
+#include "Secrets/secretmanager_p.h"
+#include "Secrets/serialization_p.h"
+
+#include <QtDBus/QDBusPendingReply>
+#include <QtDBus/QDBusPendingCallWatcher>
+
+using namespace Sailfish::Secrets;
+
+HealthCheckRequestPrivate::HealthCheckRequestPrivate()
+    : m_status(Request::Inactive)
+    , m_saltDataHealth(HealthCheckRequest::HealthUnknown)
+    , m_masterlockHealth(HealthCheckRequest::HealthUnknown)
+{
+}
+
+/*!
+ * \class HealthCheckRequest
+ * \brief Allows a client request information about the well-being of secrets data.
+ *
+ * An example of retrieving information about the secrets data health follows:
```

Comment:
Please add some more description to make it clear that this request is really intended for use by the system settings application only, and that most clients will never need to use this request.